### PR TITLE
169 service group can not be retrieved by apikey and resource

### LINKED ISSUE #169: Service Group can not be retrieved by apikey and resource
IoT Agent endpoint does not support any parameters for querying service group, see [here](https://iotagent-node-lib.readthedocs.io/en/latest/api.html#get-iotservices). Therefore, the FiliP function [get_group](https://github.com/RWTH-EBC/FiLiP/blob/526f55d2261dda0fc1dfc2780eef790b75df0140/filip/clients/ngsi_v2/iota.py#L156) does not work as expected. We can implement some workaround, e.g. filter function...

### PR DESCRIPTION
**Pull request by djs0109**
close #169

### CODE REVIEW

File: filip/clients/ngsi_v2/iota.py [hunk 1]
```diff
@@ -163,19 +164,15 @@ def get_group(self, *, resource: str, apikey: str) -> ServiceGroup:
...
-        except requests.RequestException as err:
-            self.log_error(err=err, msg=None)
-            raise
+        groups = self.get_group_list()
+        groups = filter_group_list(group_list=groups, resources=resource, apikeys=apikey)
+        if len(groups) == 1:
+            group = groups[0]
+            return group
+        elif len(groups) == 0:
+            raise KeyError(f"Service group with resource={resource} and apikey={apikey} was not found")
```
Comment (djs0109) [inline filip/clients/ngsi_v2/iota.py:173]: I am not sure with the KeyError here. Maybe there is a better Error type or even another way to inform the users.

Comment (tstorek) [reply filip/clients/ngsi_v2/iota.py:173]: Well, you can use a ValueError, but in the very end you are looking for a specific combination of keys ;)

Comment (djs0109) [reply filip/clients/ngsi_v2/iota.py:173]: Thats true. Would it be better, if we build a custom error type here? Something like "GroupNotFoundError".

Comment (tstorek) [reply filip/clients/ngsi_v2/iota.py:173]: @djs0109 I suggest to stick to the keyError

Comment (djs0109) [reply filip/clients/ngsi_v2/iota.py:173]: OK, then I will resolve this point ;)

**Review by sbanoeon — APPROVED** [review]

**Pull request merged**